contracts/container/contract.go

```diff
@@ -2259,3 +2270,147 @@ func attributeFromBytes(b []byte) (Attribute, string) {
 
 	return res, ""
 }
+
+// SetAttribute sets container attribute. The attribute must be whitelisted.
+func SetAttribute(cID interop.Hash256, name string, value []byte) {
+	var (
+		ctx     = storage.GetContext()
+		ownerID = getOwnerByID(ctx, cID)
+	)
+
+	if ownerID == nil {
+		panic(cst.NotFoundError)
+	}
+
+	if !runtime.CheckWitness(common.AlphabetAddress()) &&
+		!runtime.CheckWitness(common.WalletToScriptHash(ownerID)) {
+		panic("alphabet and owner witness check failed")
+	}
+
+	switch name {
+	case corsAttributeName:
+		setCORSAttribute(cID, value)
+	default:
+		panic("attribute is immutable")
+	}
+}
+
+// setCORSAttribute sets container CORS attribute.
+// Pass an empty list to empty the attribute.
+func setCORSAttribute(cID interop.Hash256, payload []byte) {
+	var (
+		obj  = std.JSONDeserialize(payload)
+		list = obj.([]any)
+	)
+
+	for i, item := range list {
+		validateCORSRule(i, item.(map[string]any))
+	}
+
+	var (
+		info   = GetInfo(cID)
+		exists bool
+	)
+
+	for i, attr := range info.Attributes {
+		if attr.Key == corsAttributeName {
+			exists = true
+			info.Attributes[i].Value = string(payload)
+			break
+		}
+	}
+
+	if !exists {
+		info.Attributes = append(info.Attributes, Attribute{
+			Key:   corsAttributeName,
+			Value: string(payload),
+		})
+	}
+
+	storage.Put(storage.GetContext(), append([]byte{infoPrefix}, cID...), std.Serialize(info))
```

Comment:
until #538 happened, we must also update `containerPrefix` like on creation https://github.com/nspcc-dev/neofs-contract/blob/36ea58a1353a90cbecc926d08fba04be7d446fea/contracts/container/contract.go#L613-L615